nsxt/resource_nsxt_compute_manager_test.go

```diff
@@ -0,0 +1,160 @@
+					resource.TestCheckResourceAttr(testResourceName, "description", "Terraform test compute manager"),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceNsxtComputeManager_importBasic(t *testing.T) {
+	name := getAccTestResourceName()
+	clusterName := getAccTestResourceName()
+	testResourceName := "nsxt_compute_manager.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccOnlyLocalManager(t)
+			testAccTestFabric(t)
+			testAccPreCheck(t)
+			testAccTestVCCredentials(t)
+		},
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNSXComputeManagerCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNSXComputeManagerCreateTemplate(clusterName),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNSXComputeManagerExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("NSX Compute Manager resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("NSX Compute Manager resource ID not set in resources ")
+		}
+
+		client := fabric.NewComputeManagersClient(connector)
+		obj, err := client.Get(resourceID)
+		if err != nil {
+			return fmt.Errorf("error while retrieving Compute Manager ID %s. Error: %v", resourceID, err)
+		}
+
+		if displayName == *obj.DisplayName {
+			return nil
+		}
+		return fmt.Errorf("NSX Compute Manager %s wasn't found", displayName)
+	}
+}
+
+func testAccNSXComputeManagerCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+
+	for _, rs := range state.RootModule().Resources {
+		if rs.Type != "nsxt_compute_manager" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		client := fabric.NewComputeManagersClient(connector)
+		obj, err := client.Get(resourceID)
+
+		if isNotFoundError(err) {
+			return nil
+		}
+
+		if err != nil {
+			return fmt.Errorf("error while retrieving Compute Manager ID %s. Error: %v", resourceID, err)
+		}
+
+		if obj.DisplayName != nil && displayName == *obj.DisplayName {
+			return fmt.Errorf("NSX Compute Manager %s still exists", displayName)
+		}
+	}
+
+	return nil
+}
+
+func testAccNSXComputeManagerCreateTemplate(displayName string) string {
+	return fmt.Sprintf(`
+resource "nsxt_compute_manager" "test" {
+  description  = "Terraform test compute manager"
+  display_name = "%s"
+  tag {
+    scope = "scope1"
+    tag   = "tag1"
+  }
+
+  server = "%s"
+
+  credential {
+    username_password_login {
+      username = "%s"
```

Comment:
would the API fail if credentials are incorrect? If no, I suggest to use dummy credentials here and avoid basing this test on `testAccTestVCCredentials`

Reply:
NSX validates the credentials and tests the connection. And it should be a vCenter host which isn't already connected to NSX (maybe this NSX instance?) so IDK how we run this in CI. Unless we keep an extra NSX node for that purpose.